### PR TITLE
chore: pagination enhancement

### PR DESCRIPTION
**Pull request by sagarnaikjuspay**
## Type of Change

- [ ] Bugfix
- [ ] New feature
- [x] Enhancement
- [ ] Refactoring
- [ ] Dependency updates
- [ ] Documentation
- [ ] CI/CD

## Description
New pagination design and compoment enhancement 
image

## Where to test it?
- [x] INTEG
- [x] SANDBOX
- [x] PROD

## Checklist

- [x] I ran `npm run re:build`
- [x] I reviewed submitted code
- [ ] I added unit tests for my changes where possible

### CODE REVIEW

File: src/screens/Payouts/PayoutsList.res [hunk 1]
```diff
@@ -13,9 +13,16 @@ let make = () => {
   let (filters, setFilters) = React.useState(_ => None)
   let defaultValue: LoadedTable.pageDetails = {offset: 0, resultsPerPage: 10}
   let pageDetailDict = Recoil.useRecoilValueFromAtom(LoadedTable.table_pageDetails)
+  let setPageDetails = Recoil.useSetRecoilState(LoadedTable.table_pageDetails)
   let pageDetail = pageDetailDict->Dict.get("Payouts")->Option.getOr(defaultValue)
   let (offset, setOffset) = React.useState(_ => pageDetail.offset)
 
+  let clearPageDetails = () => {
+    let newDict = pageDetailDict->Dict.toArray->Dict.fromArray
```
Comment (PritishBudhiraja) [inline src/screens/Payouts/PayoutsList.res:21]: newDict->DictionaryUtils.copyOfDict

File: src/screens/Payouts/PayoutsList.res [hunk 2]
```diff
@@ -13,9 +13,16 @@ let make = () => {
...
+    newDict->Dict.set("Payouts", defaultValue)
```
Comment (PritishBudhiraja) [inline src/screens/Payouts/PayoutsList.res:22]: And why we are manipulating a state like this? As pageDetailDict is an atomvalue it should have been done like this please try this - 
```
    setPageDetails(prev => {
      let newDict = prev->DictionaryUtils.copyOfDict
      newDict->Dict.set("Payouts", defaultValue)
      newDict
    })
```

**Review by PritishBudhiraja — CHANGES_REQUESTED** [review]
And why we are manipulating a state like this everywhere? As pageDetailDict is an atom value it should have been done like this please try this -
```
    setPageDetails(prev => {
      let newDict = prev->DictionaryUtils.copyOfDict
      newDict->Dict.set("Payouts", defaultValue)
      newDict
    })
```

File: src/components/Paginator.res [hunk 3]
```diff
@@ -1,60 +1,138 @@
...
   ~resultsPerPage,
   ~setOffset,
-  ~handleRefetch=?,
+  ~handleRefetch as _=?,
   ~currrentFetchCount,
-  ~downloadCsv=?,
+  ~downloadCsv as _=?,
   ~isNewPaginator=false,
-  ~actualData,
+  ~actualData as _,
```
Comment (PritishBudhiraja) [inline src/components/Paginator.res:121]: if not using remove it

File: src/components/Paginator.res [hunk 4]
```diff
@@ -1,60 +1,138 @@
...
+module NewPagination = {
+  @react.component
+  let make = (~resultsPerPage, ~totalResults, ~currentPage, ~paginate) => {
+    let pageNumbers = []
+    let (arrow, setArrow) = React.useState(_ => false)
+    let total = Math.ceil(Int.toFloat(totalResults) /. Int.toFloat(resultsPerPage))->Float.toInt
+
+    for x in 1 to total {
+      Array.push(pageNumbers, x)->ignore
+    }
```
Comment (PritishBudhiraja) [inline src/components/Paginator.res:31]:
```
let pageNumbers = Array.fromInitializer(~length=total, i => i + 1)
```

File: src/components/Paginator.res [hunk 5]
```diff
@@ -1,60 +1,138 @@
...
+              }}
+            </Menu.Button>
+            <TransitionWrapper>
+              <Menu.Items className=dropdownStyle>
+                {props => {
+                  if props["open"] {
+                    setArrow(_ => true)
+                  } else {
+                    setArrow(_ => false)
+                  }
```
Comment (Riddhiagrawal001) [inline src/components/Paginator.res:78]: setArrow(_=> props["open"])

File: src/components/Paginator.res [hunk 6]
```diff
@@ -63,60 +141,66 @@ let make = (
...
```
Comment (Riddhiagrawal001) [inline src/components/Paginator.res:174]: same here

File: src/screens/Order/Orders.res [hunk 7]
```diff
@@ -22,9 +22,16 @@ let make = (~previewOnly=false) => {
   let defaultValue: LoadedTable.pageDetails = {offset: 0, resultsPerPage: 20}
   let pageDetailDict = Recoil.useRecoilValueFromAtom(LoadedTable.table_pageDetails)
   let pageDetail = pageDetailDict->Dict.get("Orders")->Option.getOr(defaultValue)
+  let setPageDetails = Recoil.useSetRecoilState(LoadedTable.table_pageDetails)
   let (offset, setOffset) = React.useState(_ => pageDetail.offset)
   let {generateReport} = HyperswitchAtom.featureFlagAtom->Recoil.useRecoilValueFromAtom
 
+  let clearPageDetails = () => {
+    let newDict = pageDetailDict->Dict.toArray->Dict.fromArray
```
Comment (Riddhiagrawal001) [inline src/screens/Order/Orders.res:30]: Do we need this `->Dict.toArray->Dict.fromArray `

File: src/screens/Refunds/Refund.res [hunk 8]
```diff
@@ -14,8 +14,15 @@ let make = () => {
   let defaultValue: LoadedTable.pageDetails = {offset: 0, resultsPerPage: 20}
   let pageDetailDict = Recoil.useRecoilValueFromAtom(LoadedTable.table_pageDetails)
   let pageDetail = pageDetailDict->Dict.get("Refunds")->Option.getOr(defaultValue)
+  let setPageDetails = Recoil.useSetRecoilState(LoadedTable.table_pageDetails)
   let (offset, setOffset) = React.useState(_ => pageDetail.offset)
 
+  let clearPageDetails = () => {
+    let newDict = pageDetailDict->Dict.toArray->Dict.fromArray
```
Comment (Riddhiagrawal001) [inline src/screens/Refunds/Refund.res:21]: Same here `->Dict.toArray->Dict.fromArray `

File: src/components/Paginator.res [hunk 9]
```diff
@@ -1,60 +1,138 @@
...
+                    size=15
+                  />
+                </>
+              }}
+            </Menu.Button>
+            <TransitionWrapper>
+              <Menu.Items className=dropdownStyle>
+                {props => {
+                  if props["open"] {
+                    setArrow(_ => true)
```
Comment (JeevaRamu0104) [inline src/components/Paginator.res:75]: please check whether your getting any warning in console.Because setState is happening at every render

**Pull request closed without merging**